moment_kinetics/src/velocity_moments.jl

```diff
@@ -486,8 +486,22 @@ end
 
 """
 calculate the updated density (dens) and parallel pressure (ppar) for all species
-"""
-function update_moments!(moments, ff, vpa, vperp, z, r, composition)
+this function is only used once after initialisation
+the function used to update moments at run time is update_derived_moments! in time_advance.jl
+"""
+function update_moments!(moments, ff_in, gyroavs::gyro_operators, vpa, vperp, z, r, composition,
+        r_spectral, geometry, scratch_dummy, z_advect)
+    ff = scratch_dummy.buffer_vpavperpzrs_1 # the buffer array for the charged pdf -> make sure not to reuse this array below
+    if composition.gyrokinetic_ions
+        # fill buffer with ring-averaged F (gyroaverage at fixed position)
+        gyroaverage_pdf!(ff,ff_in,gyroavs,vpa,vperp,z,r,composition)
+    else
+        # copy F into buffer (drift-kinetic)
+        begin_s_r_z_vperp_vpa_region()
+        @loop_s_r_z_vperp_vpa is ir iz ivperp ivpa begin 
+            ff[ivpa,ivperp,iz,ir,is] = ff_in[ivpa,ivperp,iz,ir,is]
+        end
+    end
```

Comment:
Skip copy when not gyroaveraging
```suggestion
    if composition.gyrokinetic_ions
        ff = scratch_dummy.buffer_vpavperpzrs_1 # the buffer array for the charged pdf -> make sure not to reuse this array below
        # fill buffer with ring-averaged F (gyroaverage at fixed position)
        gyroaverage_pdf!(ff,ff_in,gyroavs,vpa,vperp,z,r,composition)
    else
        ff = ff_in
    end
```